tasks/upgrade.js

```diff
@@ -0,0 +1,22 @@
+const { task } = require("hardhat/config");
+
+task("upgrade", "upgrade a contract").setAction(async () => {
+  await hre.run("compile");
+
+  if (
+    network.name !== "mumbai" &&
+    network.name !== "matic" &&
+    network.name !== "goerli"
+  ) {
```

Comment:
```suggestion
  const supportedChains = ['mumbai', 'matic', 'goerli']
  
  if (!supportedChains.includes(network.name)) {
```
I think we can clean this up a tad